pennylane/tape/tapes/qubit_param_shift.py

```diff
@@ -167,9 +169,7 @@ def processing_fn(results):
                 array[float]: 1-dimensional array of length determined by the tape output
                 measurement statistics
             """
-            res_forward = np.array(results[0])
-            res_backward = np.array(results[1])
-            return (res_forward - res_backward) / (2 * np.sin(s))
+            return np.dot(coeffs, results)

```

Comment:
@antalszava, I had to modify the `QubitParamShift` tape, I think it was missed in the previous PR :)

I noticed while modifying it that we can simplify some of the logic (for example, using `np.dot`). Worth backporting to the other methods done in #909?

Reply:
Oh, thanks for catching that @josh146! Sure, I'll look into the other cases where it could be simplified